backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/controller/StockViewController.java

```diff
@@ -87,75 +90,80 @@ public class StockViewController {
+
     @GetMapping("materials")
     @ResponseBody
     @Operation(description = "Returns a list of all materials (excluding products)")
+    @CrossOrigin
     public List<FrontendMaterialDto> getMaterials() {
         return materialService.findAllMaterials()
             .stream()
             .map(mat -> new FrontendMaterialDto(mat.getOwnMaterialNumber(), mat.getName()))
             .collect(Collectors.toList());
     }
 
-    @CrossOrigin
     @GetMapping("materialnumbers-mapping")
-    @ResponseBody
     @Operation(description = "Returns a mapping of all material numbers, that others partners are using" +
         "for the material given in the request parameter.")
-    @ApiResponses(value = {@ApiResponse(content = @Content(examples = {
-        @ExampleObject(name = "Basic sample", value = "{" +
-            "  \"BPNL1234567890ZZ\": \"MNR-8101-ID146955.001\"," +
-            "  \"BPNL4444444444XX\": \"MNR-7307-AU340474.002\"}")
-    }))})
-    public Map<String, String> getMaterialNumbers(@RequestParam String ownMaterialNumber) {
-        return mprService.getBPNL_To_MaterialNumberMap(ownMaterialNumber);
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", content = @Content(examples = {
+            @ExampleObject(name = "Basic sample", value = "{" +
+                "  \"BPNL1234567890ZZ\": \"MNR-8101-ID146955.001\"," +
+                "  \"BPNL4444444444XX\": \"MNR-7307-AU340474.002\"}")})),
+        @ApiResponse(responseCode = "400", description = "Invalid parameter")})
+    @CrossOrigin
+    public ResponseEntity<Map<String, String>> getMaterialNumbers(@RequestParam String ownMaterialNumber) {
+        if (!materialPattern.matcher(ownMaterialNumber).matches()) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(400));
+        }
+        return ResponseEntity.ok(mprService.getBPNL_To_MaterialNumberMap(ownMaterialNumber));
     }
 
-    @CrossOrigin
     @GetMapping("products")
     @ResponseBody
     @Operation(description = "Returns a list of all products (excluding materials)")
+    @CrossOrigin
     public List<FrontendMaterialDto> getProducts() {
         return materialService.findAllProducts()
             .stream()
             .map(mat -> new FrontendMaterialDto(mat.getOwnMaterialNumber(), mat.getName()))
             .collect(Collectors.toList());
     }
 
-    @CrossOrigin
     @GetMapping("product-stocks")
     @ResponseBody
     @Operation(description = "Returns a list of all product-stocks")
+    @CrossOrigin
     public List<ProductStockDto> getProductStocks() {
         return productStockService.findAll().stream()
             .map(this::convertToDto)
             .collect(Collectors.toList());
     }
 
-    @CrossOrigin
     @PostMapping("product-stocks")
     @ResponseBody
     @Operation(description = "Creates a new product-stock")
+    @CrossOrigin
     public ProductStockDto createProductStocks(@RequestBody ProductStockDto productStockDto) {
```

Comment:
The product stock relies on manual user input. Please also add pattern checks on the dto.